src/components/omni-panel-status-labels.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { Typography, makeStyles } from '@material-ui/core';
+
+export interface LabelProps {
+  modalLabelClass: string;
+  modeText?: string;
+  name?: string;
+  hideTextStyleOverride?: string;
+}
+
+export const useStyles = makeStyles(theme => ({
+  hideText: {
+    overflow: "hidden",
+    textOverflow: "ellipsis",
+    whiteSpace: "nowrap",
+    maxWidth: "10rem",
+  }
+}));
+
+export default function OmniPanelStatusLabels(props: LabelProps) {
+  const classes = useStyles();
+  const { modalLabelClass, modeText, name, hideTextStyleOverride } = props;
+
+  return (
+    <React.Fragment>
+      <Typography variant="h6" className={hideTextStyleOverride ? hideTextStyleOverride : classes.hideText}>
```

Comment:
Just curious, is this the right way to do style overriding? Seems like I am violating one of the rules of presentational components again

Reply:
Simple answer: yes.

[Not useful for now] Extra information for the future
It's likely that in the future we'll want to have some components with a part of the style immutable and that only certain things such as colors, etc. can be overwritten. In that case, we shouldn't replace the entire style as we are doing now and we should probably also check the entered text type with custom Typescript types.